platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/ResolveIvyFactory.java

```diff
@@ -143,22 +147,44 @@ public ComponentResolvers create(
             if (baseRepository.isDynamicResolveMode()) {
                 moduleComponentRepository = new IvyDynamicResolveModuleComponentRepository(moduleComponentRepository, moduleResolveStateFactory);
             }
+
             moduleComponentRepository = new ErrorHandlingModuleComponentRepository(moduleComponentRepository, repositoryBlacklister);
-            moduleComponentRepository = filterRepository(repository, moduleComponentRepository, resolveContextName, consumerAttributes);
-            moduleComponentRepository = dependencyVerificationOverride.overrideDependencyVerification(moduleComponentRepository, resolveContextName, resolutionStrategy);
+            moduleComponentRepository = filterRepository(repository, moduleComponentRepository);
+            moduleComponentRepository = applyDependencyVerification(moduleComponentRepository, resolutionStrategy.isDependencyVerificationEnabled());
+
             moduleResolver.add(moduleComponentRepository);
             parentModuleResolver.add(moduleComponentRepository);
         }
 
         return moduleResolver;
     }
 
-    private ModuleComponentRepository<ModuleComponentGraphResolveState> filterRepository(ResolutionAwareRepository repository, ModuleComponentRepository<ModuleComponentGraphResolveState> moduleComponentRepository, String consumerName, AttributeContainer consumerAttributes) {
+    private static ModuleComponentRepository<ModuleComponentGraphResolveState> filterRepository(
+        ResolutionAwareRepository repository,
+        ModuleComponentRepository<ModuleComponentGraphResolveState> moduleComponentRepository
+    ) {
         Action<? super ArtifactResolutionDetails> filter = Actions.doNothing();
         if (repository instanceof ContentFilteringRepository) {
             filter = ((ContentFilteringRepository) repository).getContentFilter();
         }
-        return FilteredModuleComponentRepository.of(moduleComponentRepository, filter, consumerName, consumerAttributes);
+
+        if (filter == Actions.doNothing()) {
+            return moduleComponentRepository;
+        }
+
+        return new FilteredModuleComponentRepository(moduleComponentRepository, filter);
+    }
+
+    private ModuleComponentRepository<ModuleComponentGraphResolveState> applyDependencyVerification(
+        ModuleComponentRepository<ModuleComponentGraphResolveState> moduleComponentRepository,
+        boolean dependencyVerificationEnabled
+    ) {
+        if (!dependencyVerificationEnabled) {
+            LOGGER.warn("Dependency verification has been disabled.");
```

Comment:
💅 A test for this new warning would be good.

Reply:
This is an existing warning. It has just been moved and the text slightly changed. There are already tests that verify this warning is printed and they have been updated with the new text